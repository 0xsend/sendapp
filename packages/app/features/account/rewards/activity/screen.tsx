import {
  YStack,
  H1,
  Paragraph,
  XStack,
  Button,
  Image,
  LinearGradient,
  Stack,
  Spinner,
  Select,
  H3,
  Adapt,
  Sheet,
  type SelectItemProps,
  Theme,
} from '@my/ui'
import { ChevronDown, ChevronUp, Dot } from '@tamagui/lucide-icons'
import { IconX } from 'app/components/icons'
import { useRewardsScreenParams } from 'app/routers/params'
import { useMonthlyDistributions } from 'app/utils/distributions'
import { useId, useState } from 'react'

export function ActivityRewardsScreen() {
  const [queryParams, setRewardsScreenParams] = useRewardsScreenParams()
  const { data: distributions, isLoading } = useMonthlyDistributions()
  const [isOpen, setIsOpen] = useState(false)
  const id = useId()

  const [selectedDistributionIndex, setSelectedDistributionIndex] = useState(
    queryParams.distribution ?? 0
  )

  if (isLoading)
    return (
      <YStack f={1} pb={'$2'} pt={'$6'} gap={'$7'}>
        <Header />
        <Stack w="100%" f={1} jc={'center'} ai={'center'}>
          <Spinner color="$color" size="large" />
        </Stack>
      </YStack>
    )
  if (!distributions || !distributions[selectedDistributionIndex])
    return (
      <YStack f={1} pb={'$2'} pt={'$6'} gap={'$7'}>
        <Header />
        <Stack w="100%" f={1} jc={'center'} ai={'center'}>
          <Paragraph color={'$color10'} size={'$5'}>
            No distribution found
          </Paragraph>
        </Stack>
      </YStack>
    )

  const distributionDates = distributions.map(
    (d) =>
      `${d.qualification_end.toLocaleString('default', {
        month: 'long',
      })} ${d.qualification_end.toLocaleString('default', { year: 'numeric' })}`
  )

  const onValueChange = (value: string) => {
    setSelectedDistributionIndex(Number(value))
    setRewardsScreenParams(
      { distribution: distributions[Number(value)]?.number },
      { webBehavior: 'replace' }
    )
  }

  return (
    <YStack f={1} pb={'$2'} pt={'$6'} gap={'$7'}>
      <Header />
      <XStack w={'100%'} jc={'space-between'} ai={'center'}>
        <H3 fontWeight={'600'} color={'$color12'}>
          {`${distributionDates[selectedDistributionIndex]} Rewards`}
        </H3>
        <Select
          native={false}
          id={id}
          value={queryParams.distribution?.toString() ?? '0'}
          onValueChange={onValueChange}
          onOpenChange={setIsOpen}
          defaultValue="0"
          open={isOpen}
        >
          <Select.Trigger
            testID={'SelectDistributionDate'}
            br="$3"
            w={'fit-content'}
            borderWidth={1.5}
            theme={'green'}
            $theme-light={{ boc: '$color12', bc: isOpen ? '$color12' : '$transparent' }}
            $theme-dark={{
              boc: '$color4',
              bc: isOpen ? '$color4' : '$transparent',
            }}
            iconAfter={
              isOpen ? (
                <ChevronUp
                  $theme-dark={{
                    color: '$color0',
                  }}
                  color={'$color11'}
                />
              ) : (
                <ChevronDown
                  $theme-dark={{
                    color: '$color12',
                  }}
                  color="$black"
                />
              )
            }
          >
            <Select.Value
              testID={'SelectDistributionDateValue'}
              fontWeight={'bold'}
              color={'$black'}
              $theme-dark={{
                color: isOpen ? '$color0' : '$color12',
              }}
              placeholder={distributions[selectedDistributionIndex]?.number}
            />
          </Select.Trigger>

          <Adapt when="sm" platform="touch">
            <Sheet native modal dismissOnSnapToBottom snapPoints={[30]} animation={'quick'}>
              <Sheet.Frame maw={738} bc={'$color1'}>
                <Sheet.Handle py="$5" f={1} bc="transparent" jc={'space-between'} opacity={1} m={0}>
                  <XStack ai="center" jc="space-between" w="100%" px="$4">
                    <Paragraph fontSize={'$5'} fontWeight={'700'} color={'$color12'}>
                      Select Distribution
                    </Paragraph>
                    <Button
                      chromeless
                      unstyled
                      icon={<IconX color={'$color12'} size={'$1.5'} />}
                      onPress={() => setIsOpen(false)}
                    />
                  </XStack>
                </Sheet.Handle>
                <Sheet.ScrollView>
                  <Adapt.Contents />
                </Sheet.ScrollView>
              </Sheet.Frame>
              <Sheet.Overlay />
            </Sheet>
          </Adapt>

          <Select.Content zIndex={200000}>
            <Select.Viewport br={'$3'} w={320} btrr={0} boc="transparent" bc={'$color1'}>
              <Select.Group>
                {distributions.map((distribution, i) => (
                  <DistributionItem
                    isActive={
                      distribution.number === distributions[selectedDistributionIndex]?.number
                    }
                    value={i.toString()}
                    index={i}
                    key={distribution?.number}
                  >
                    {distributionDates[i]}
                  </DistributionItem>
                ))}
              </Select.Group>
            </Select.Viewport>
          </Select.Content>
        </Select>
      </XStack>
    </YStack>
  )
}

const Header = () => (
  <Stack w={'100%'} h={224} position="relative" jc={'center'} br={'$6'} overflow="hidden">
    <Image
      pos={'absolute'}
      br={'$6'}
      t={0}
      zIndex={0}
      source={{
        height: 1024,
        width: 1024,
        uri: 'https://ghassets.send.app/app_images/flower.jpg',
      }}
      h={'100%'}
      w={'100%'}
      $sm={{
        scale: 1.5,
      }}
      objectFit="cover"
    />
    <LinearGradient
      pos={'absolute'}
      br={'$6'}
      t={0}
      start={[0, 0]}
      end={[0, 1]}
      fullscreen
      colors={['$darkest', 'transparent', '$darkest']}
    />

    <YStack p="$4" pt={'$3'} maw={463} position="absolute" zIndex={1}>
      <H1 tt={'uppercase'} $theme-light={{ col: '$color0' }} $sm={{ size: '$9' }}>
        Unlock <br />
        Extra Rewards
      </H1>
      <Paragraph
        $theme-light={{ col: '$color2' }}
        color={'$color10'}
        size={'$5'}
        $sm={{ size: '$2' }}
      >
        Register at least 1 Sendtag, maintain the minimum balance, avoid selling, and refer others
        for a bonus multiplier.
      </Paragraph>
    </YStack>
  </Stack>
)

const DistributionItem = ({
  isActive,
  value,
  index,
  children,
  ...props
}: {
  isActive: boolean
} & SelectItemProps) => {
  return (
    <Select.Item index={index} value={value} bc="transparent" f={1} w="100%" {...props}>
      <XStack gap={'$1'} $gtLg={{ gap: '$3.5' }} w={'100%'} ai={'center'}>
        <Select.ItemText
          display="flex"
          fontSize={'$5'}
          fontWeight={'500'}
          textTransform={'uppercase'}
          color={'$color12'}
          f={1}
          jc={'center'}
          ai={'center'}
        >
          {children}
        </Select.ItemText>
        {isActive && (
          <Theme name="green_active">
            <Dot size={'$3'} />
          </Theme>
        )}
      </XStack>
    </Select.Item>
  )
}
