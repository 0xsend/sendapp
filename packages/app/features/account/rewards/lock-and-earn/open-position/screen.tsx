import { Button, Card, H1, Paragraph, XStack, YStack, SubmitButton, Stack } from '@my/ui'
import { z } from 'zod'
import { SchemaForm, formFields } from 'app/utils/SchemaForm'
import { FormProvider, useForm } from 'react-hook-form'
import type { PropsWithChildren } from 'react'
import { coins, coinsDict, sendCoin } from 'app/data/coins'
import { IconCoin } from 'app/components/icons/IconCoin'
import { useSendAccountBalances } from 'app/utils/useSendAccountBalances'
import { formatUnits } from 'viem'
import formatAmount from 'app/utils/formatAmount'

const removeDuplicateInString = (text: string, substring: string) => {
  const [first, ...after] = text.split(substring)
  return first + (after.length ? `${substring}${after.join('')}` : '')
}

export function OpenPosition() {
  return (
    <YStack pt={'$size.3.5'} $gtLg={{ pt: 0 }} f={1} $gtMd={{ ml: '$4' }}>
      <YStack pb={'$size.3.5'} gap="$size.0.9">
        <H1 size={'$9'} fontWeight={'900'} color="$color12" tt={'uppercase'}>
          Open a new position
        </H1>
        <Paragraph color={'$color10'} size={'$5'} maw={650}>
          Liquidity Providers get to share 1.5% of all swaps (added to your rewards automatically),
          and 5b in $SEND rewards over 12 months.
        </Paragraph>
      </YStack>

      <YStack $gtMd={{ fd: 'row' }} gap={'$size.1.5'}>
        <OpenPositionForm />
      </YStack>
    </YStack>
  )
}

const OpenPositionForm = () => {
  const form = useForm<z.infer<typeof OpenPositionSchema>>()
  const { balances } = useSendAccountBalances()

  const formToken = form.watch('token')
  const formTokenAmount = form.watch('tokenAmount')
  const formSendAmount = form.watch('sendAmount')

  const [tokenSymbol, tokenDecimals] = [
    coinsDict[formToken]?.symbol,
    coinsDict[formToken]?.decimals,
  ]

  const sendBalance = (
    balances?.SEND ? formatUnits(balances.SEND, sendCoin.decimals) : 0n
  ).toString()
  const tokenBalance = formatUnits(balances?.[tokenSymbol] ?? 0n, tokenDecimals)

  const canSubmit =
    formTokenAmount &&
    Number(formTokenAmount) > 0 &&
    Number(formTokenAmount) <= Number(tokenBalance) &&
    formSendAmount &&
    Number(formSendAmount) > 0 &&
    Number(formSendAmount) <= Number(sendBalance)

  return (
    <FormProvider {...form}>
      <SchemaForm
        form={form}
        schema={OpenPositionSchema}
        onSubmit={() => {}}
        props={{
          tokenAmount: {
            ...sharedStyles,
            onChangeText: (text) => {
              const formattedText = removeDuplicateInString(text, '.').replace(/[^0-9.]/g, '') //remove duplicate "." then filter out any letters
              form.setValue('tokenAmount', formattedText)
            },
          },
          sendAmount: {
            ...sharedStyles,
            inputMode: 'numeric',
            onChangeText: (text) => {
              const formattedText = text.replace(/[^0-9]/g, '') // only accept numbers
              form.setValue('sendAmount', formattedText)
            },
          },
          token: {
            supportedCoins: coins.filter(({ symbol }) => symbol !== 'SEND'),
          },
        }}
        defaultValues={{
          token: 'eth',
        }}
        formProps={{
          testID: 'OpenPositionForm',
          $gtSm: { maxWidth: '100%' },
          jc: 'flex-start',
          als: 'flex-start',
          f: 1,
          height: '100%',
        }}
        renderAfter={({ submit }) => (
          <XStack mt={'$size.3.5'}>
            <SubmitButton
              variant={!canSubmit ? 'outlined' : undefined}
              theme={canSubmit ? 'green' : 'dim'}
              borderRadius={'$4'}
              disabled={!canSubmit}
              // @TODO: submit
              onPress={() => console.log('press')}
            >
              <Button.Text
                tt={'uppercase'}
                color={canSubmit ? '$color1' : '$color12'}
                fontWeight={500}
                ff="$mono"
              >
                {canSubmit ? 'Provide Liquidity' : 'Enter USDC or SEND Amount'}
              </Button.Text>
            </SubmitButton>
          </XStack>
        )}
      >
        {({ token, tokenAmount, sendAmount }) => (
          <YStack $gtMd={{ fd: 'row' }} gap={'$size.1.5'}>
            <FormCard
              balance={tokenBalance}
              onMaxPress={() => form.setValue('tokenAmount', tokenBalance)}
            >
              <Stack w={'$size.3.5'}>
                {coinsDict[formToken] && (
                  <IconCoin coin={coinsDict[formToken]} size={'$size.3.5'} />
                )}
              </Stack>
              {tokenAmount}
              {token}
            </FormCard>

            <FormCard
              balance={sendBalance}
              onMaxPress={() => form.setValue('sendAmount', sendBalance)}
            >
              <Stack>
                <IconCoin coin={sendCoin} size={'$size.3.5'} />
              </Stack>
              {sendAmount}
              <Paragraph fontWeight={600} fontSize={'$8'}>
                SEND
              </Paragraph>
            </FormCard>
          </YStack>
        )}
      </SchemaForm>
    </FormProvider>
  )
}

const OpenPositionSchema = z.object({
  token: formFields.coin,
  tokenAmount: formFields.text,
  sendAmount: formFields.text,
})

const FormCard = ({
  balance,
  onMaxPress,
  children,
}: {
  balance: string
  onMaxPress: () => void
} & PropsWithChildren) => {
  return (
    <Card p={'$size.3.5'} flex={1}>
      <XStack
        pb="$size.0.9"
        borderBottomColor={'$color10'}
        borderBottomWidth={1}
        justifyContent="space-between"
        alignItems="center"
        gap="$size.0.9"
      >
        {children}
      </XStack>

      <XStack pt="$size.0.9" justifyContent="space-between">
        <XStack gap="$size.0.75">
          <Paragraph color="$color10" size={'$5'}>
            Balance:
          </Paragraph>
          <Paragraph size={'$5'} fontWeight={'600'}>
            {formatAmount(balance, 9, 2)}
          </Paragraph>
        </XStack>
        <Button unstyled onPress={onMaxPress}>
          <Button.Text size={'$5'} color="$color10" textDecorationLine="underline">
            MAX
          </Button.Text>
        </Button>
      </XStack>
    </Card>
  )
}

const sharedStyles = {
  fieldsetProps: { flex: 1 },
  backgroundColor: '$color1',
  flex: 1,
  inputMode: 'decimal',
  fontSize: '$9',
  ff: '$mono',
  placeholder: '0',
  p: 0,
  lineHeight: '$1',
  borderColor: 'transparent',
  outlineColor: 'transparent',
  fontStyle: 'normal',
  '$theme-dark': {
    placeholderTextColor: '$color10',
  },
  $sm: {
    w: '100%',
  },
} as const
