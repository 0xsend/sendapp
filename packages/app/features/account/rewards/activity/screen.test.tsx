import { Wrapper } from 'app/utils/__mocks__/Wrapper'
import { ActivityRewardsScreen } from './screen'
import { act, render, screen } from '@testing-library/react-native'

jest.mock('app/utils/distributions', () => ({
  useMonthlyDistributions: () => ({
    data: [
      {
        number: 7,
        chain_id: 845337,
        qualification_end: Date.UTC(2024, 6, 15),
        distribution_shares: [
          {
            amount: 1,
            index: 1,
          },
        ],
        distribution_verifications_summary: [
          {
            tag_referrals: 123,
          },
        ],
      },
    ],
  }),
}))

jest.mock('app/routers/params', () => ({
  useRewardsScreenParams: () => [{ distributionNumber: 1 }, jest.fn()],
}))

describe('ActivityRewardsScreen', () => {
  it('renders', async () => {
    jest.useFakeTimers()
    jest.setSystemTime(Date.UTC(2024, 6, 12))
    render(
      <Wrapper>
        <ActivityRewardsScreen />
      </Wrapper>
    )

    await act(async () => {
      jest.advanceTimersByTime(2000)
    })
    expect(screen.getByTestId('SelectDistributionDate')).toBeVisible()
    expect(screen.toJSON()).toMatchSnapshot('ActivityRewardsScreen')
  })
})
