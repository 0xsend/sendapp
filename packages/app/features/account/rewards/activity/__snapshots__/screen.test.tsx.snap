// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityRewardsScreen renders: ActivityRewardsScreen 1`] = `
[
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "column",
        "gap": 39,
        "paddingBottom": 7,
        "paddingTop": 32,
      }
    }
  >
    <View
      style={
        {
          "borderBottomLeftRadius": 16,
          "borderBottomRightRadius": 16,
          "borderTopLeftRadius": 16,
          "borderTopRightRadius": 16,
          "height": 224,
          "justifyContent": "center",
          "overflow": "hidden",
          "position": "relative",
          "width": "100%",
        }
      }
    >
      <Image
        objectFit="cover"
        source={
          {
            "height": 1024,
            "uri": "https://ghassets.send.app/app_images/flower.jpg",
            "width": 1024,
          }
        }
        style={
          {
            "borderBottomLeftRadius": 16,
            "borderBottomRightRadius": 16,
            "borderTopLeftRadius": 16,
            "borderTopRightRadius": 16,
            "height": "100%",
            "position": "absolute",
            "top": 0,
            "transform": [
              {
                "scale": 1.5,
              },
            ],
            "width": "100%",
            "zIndex": 0,
          }
        }
      />
      <View
        style={
          {
            "borderBottomLeftRadius": 16,
            "borderBottomRightRadius": 16,
            "borderTopLeftRadius": 16,
            "borderTopRightRadius": 16,
            "bottom": 0,
            "flexDirection": "column",
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <ViewManagerAdapter_ExpoLinearGradient
          colors={
            [
              4278720025,
              0,
              4278720025,
            ]
          }
          endPoint={
            [
              0,
              1,
            ]
          }
          startPoint={
            [
              0,
              0,
            ]
          }
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        />
      </View>
      <View
        style={
          {
            "flexDirection": "column",
            "maxWidth": 463,
            "paddingBottom": 18,
            "paddingLeft": 18,
            "paddingRight": 18,
            "paddingTop": 13,
            "position": "absolute",
            "zIndex": 1,
          }
        }
      >
        <Text
          accessibilityRole="header"
          style={
            {
              "color": "#FFFFFF",
              "fontFamily": "System",
              "fontSize": 30,
              "fontWeight": "700",
              "lineHeight": 34,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "textTransform": "none",
              "userSelect": "auto",
            }
          }
          suppressHighlighting={true}
        >
          Unlock 
          <br />
          Extra Rewards
        </Text>
        <Text
          style={
            {
              "color": "#B2B2B2",
              "fontFamily": "System",
              "fontSize": 14.399999999999999,
              "fontWeight": "300",
              "lineHeight": 17,
              "userSelect": "auto",
            }
          }
          suppressHighlighting={true}
        >
          Register at least 1 Sendtag, maintain the minimum balance, avoid selling, and refer others for a bonus multiplier.
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <Text
        accessibilityRole="header"
        style={
          {
            "color": "#FFFFFF",
            "fontFamily": "System",
            "fontSize": 23,
            "fontWeight": "600",
            "lineHeight": 27,
            "marginBottom": 0,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 0,
            "textTransform": "none",
            "userSelect": "auto",
          }
        }
        suppressHighlighting={true}
      >
        1,721,001,600,000 1,721,001,600,000 Rewards
      </Text>
      <View
        aria-autocomplete="none"
        aria-expanded={false}
        cancelable={true}
        disabled={false}
        focusVisibleStyle={
          {
            "outlineColor": "$outlineColor",
            "outlineStyle": "solid",
            "outlineWidth": 2,
          }
        }
        focusable={[Function]}
        id=":r0:"
        minPressDuration={0}
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onPress={[Function]}
        onPressIn={[Function]}
        onPressOut={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#081619",
            "borderBottomColor": "#40FB50",
            "borderBottomLeftRadius": 7,
            "borderBottomRightRadius": 7,
            "borderBottomWidth": 1.5,
            "borderLeftColor": "#40FB50",
            "borderLeftWidth": 1.5,
            "borderRightColor": "#40FB50",
            "borderRightWidth": 1.5,
            "borderStyle": "solid",
            "borderTopColor": "#40FB50",
            "borderTopLeftRadius": 7,
            "borderTopRightRadius": 7,
            "borderTopWidth": 1.5,
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "justifyContent": "space-between",
            "maxWidth": "100%",
            "minHeight": 44,
            "overflow": "hidden",
            "paddingBottom": 10,
            "paddingLeft": 18,
            "paddingRight": 18,
            "paddingTop": 10,
            "width": "fit-content",
          }
        }
        testID="SelectDistributionDate"
        type="button"
      >
        <Text
          lineBreakMode="clip"
          numberOfLines={1}
          pointerEvents="none"
          style={
            {
              "color": "#FFFFFF",
              "fontFamily": "System",
              "fontSize": 16.8,
              "fontWeight": "bold",
              "lineHeight": 19,
              "userSelect": "none",
            }
          }
          suppressHighlighting={true}
          testID="SelectDistributionDateValue"
        >
          1,721,001,600,000 1,721,001,600,000
        </Text>
        <View
          pointerEvents="none"
          style={
            {
              "height": 16.8,
              "minHeight": 16.8,
              "minWidth": 16.8,
              "width": 16.8,
            }
          }
        />
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={16.8}
          bbWidth={16.8}
          fill="none"
          focusable={false}
          height={16.8}
          meetOrSlice={0}
          minX={0}
          minY={0}
          stroke="#FFFFFF"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "color": "#FFFFFF",
              },
              {
                "flex": 0,
                "height": 16,
                "width": 16,
              },
            ]
          }
          tintColor="#FFFFFF"
          vbHeight={24}
          vbWidth={24}
          width={16.8}
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          >
            <RNSVGPath
              d="m6 9 6 6 6-6"
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              propList={
                [
                  "stroke",
                ]
              }
              stroke={
                {
                  "payload": 4294967295,
                  "type": 0,
                }
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>,
  <View
    pointerEvents="box-none"
    style={
      {
        "bottom": 0,
        "flexDirection": "column",
        "left": 0,
        "maxWidth": "100%",
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 100001,
      }
    }
  >
    <View
      animation="quick"
      collapsable={false}
      disableClassName={true}
      forwardedRef={[Function]}
      jestAnimatedStyle={
        {
          "value": {
            "transform": [
              {
                "translateY": 10000.1,
              },
            ],
          },
        }
      }
      onLayout={[Function]}
      style={
        {
          "height": "30%",
          "minHeight": "30%",
          "opacity": 0,
          "pointerEvents": "none",
          "position": "absolute",
          "transform": [
            {
              "translateY": 10000.1,
            },
          ],
          "width": "100%",
          "zIndex": 100001,
        }
      }
    >
      <View
        pointerEvents="none"
        style={
          {
            "backgroundColor": "#111f22",
            "borderTopLeftRadius": 16,
            "borderTopRightRadius": 16,
            "flex": 1,
            "flexDirection": "column",
            "height": 0,
            "maxHeight": "100%",
            "maxWidth": 738,
            "overflow": "hidden",
            "width": "100%",
          }
        }
      >
        <View
          cancelable={true}
          disabled={false}
          focusable={true}
          minPressDuration={0}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onPress={[Function]}
          onPressIn={[Function]}
          onPressOut={[Function]}
          pointerEvents="none"
          style={
            {
              "backgroundColor": "transparent",
              "borderBottomLeftRadius": 100,
              "borderBottomRightRadius": 100,
              "borderTopLeftRadius": 100,
              "borderTopRightRadius": 100,
              "flex": 1,
              "flexDirection": "row",
              "height": 10,
              "justifyContent": "space-between",
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "opacity": 1,
              "paddingBottom": 24,
              "paddingTop": 24,
              "zIndex": 10,
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingLeft": 18,
                "paddingRight": 18,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#FFFFFF",
                  "fontFamily": "System",
                  "fontSize": 19.2,
                  "fontWeight": "700",
                  "lineHeight": 19,
                  "userSelect": "auto",
                }
              }
              suppressHighlighting={true}
            >
              Select Distribution
            </Text>
            <View
              cancelable={true}
              disabled={false}
              focusable={[Function]}
              minPressDuration={0}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onPress={[Function]}
              onPressIn={[Function]}
              onPressOut={[Function]}
              role="button"
              style={
                {
                  "backgroundColor": "transparent",
                  "borderBottomColor": "transparent",
                  "borderLeftColor": "transparent",
                  "borderRightColor": "transparent",
                  "borderTopColor": "transparent",
                  "flexDirection": "column",
                  "shadowColor": "rgb(0,0,0)",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={24}
                bbWidth={24}
                color="#FFFFFF"
                focusable={false}
                height={24}
                meetOrSlice={0}
                minX={0}
                minY={0}
                strokeWidth="2"
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "color": "#FFFFFF",
                    },
                    {
                      "flex": 0,
                      "height": 24,
                      "width": 24,
                    },
                  ]
                }
                tintColor="#FFFFFF"
                vbHeight={32}
                vbWidth={32}
                width={24}
              >
                <RNSVGGroup
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "strokeWidth",
                    ]
                  }
                  strokeWidth="2"
                >
                  <RNSVGPath
                    d="M8.30006 25.1004L6.90005 23.7004L14.6001 16.0004L6.90005 8.30039L8.30006 6.90039L16.0001 14.6004L23.7001 6.90039L25.1001 8.30039L17.4001 16.0004L25.1001 23.7004L23.7001 25.1004L16.0001 17.4004L8.30006 25.1004Z"
                    fill={
                      {
                        "type": 2,
                      }
                    }
                    propList={
                      [
                        "fill",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </View>
        </View>
        <RCTScrollView
          onMoveShouldSetResponder={[Function]}
          onResponderRelease={[Function]}
          onScroll={[Function]}
          onStartShouldSetResponder={[Function]}
          scrollEnabled={true}
          scrollEventThrottle={8}
          style={
            {
              "flex": 1,
            }
          }
        >
          <View>
            <View
              aria-labelledby=":r4:"
              role="group"
              style={
                {
                  "flexDirection": "column",
                  "width": "100%",
                }
              }
            >
              <View
                aria-labelledby=":r5:"
                aria-selected={true}
                cancelable={true}
                disabled={false}
                focusVisibleStyle={
                  {
                    "outlineColor": "$outlineColor",
                    "outlineStyle": "solid",
                    "outlineWidth": 1,
                  }
                }
                focusable={true}
                minPressDuration={0}
                onBlur={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onPress={[Function]}
                onPressIn={[Function]}
                onPressOut={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "borderBottomColor": "#081619",
                    "borderLeftColor": "#081619",
                    "borderRightColor": "#081619",
                    "borderTopColor": "#081619",
                    "flex": 1,
                    "flexDirection": "row",
                    "flexWrap": "nowrap",
                    "justifyContent": "space-between",
                    "maxWidth": "100%",
                    "minHeight": 44,
                    "overflow": "hidden",
                    "paddingBottom": 10,
                    "paddingLeft": 18,
                    "paddingRight": 18,
                    "paddingTop": 10,
                    "width": "100%",
                  }
                }
                tabIndex={-1}
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "gap": 2,
                      "width": "100%",
                    }
                  }
                >
                  <Text
                    id=":r5:"
                    lineBreakMode="clip"
                    numberOfLines={1}
                    style={
                      {
                        "alignItems": "center",
                        "color": "#FFFFFF",
                        "display": "flex",
                        "flex": 1,
                        "fontFamily": "System",
                        "fontSize": 19.2,
                        "fontWeight": "500",
                        "justifyContent": "center",
                        "lineHeight": 19,
                        "textTransform": "uppercase",
                        "userSelect": "none",
                      }
                    }
                    suppressHighlighting={true}
                  >
                    1,721,001,600,000 1,721,001,600,000
                  </Text>
                  <RNSVGSvgView
                    align="xMidYMid"
                    bbHeight={36}
                    bbWidth={36}
                    fill="none"
                    focusable={false}
                    height={36}
                    meetOrSlice={0}
                    minX={0}
                    minY={0}
                    stroke="#40FB50"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth="2"
                    style={
                      [
                        {
                          "backgroundColor": "transparent",
                          "borderWidth": 0,
                        },
                        {},
                        {
                          "flex": 0,
                          "height": 36,
                          "width": 36,
                        },
                      ]
                    }
                    vbHeight={24}
                    vbWidth={24}
                    width={36}
                  >
                    <RNSVGGroup
                      fill={null}
                      propList={
                        [
                          "fill",
                          "stroke",
                          "strokeWidth",
                          "strokeLinecap",
                          "strokeLinejoin",
                        ]
                      }
                      stroke={
                        {
                          "payload": 4282448720,
                          "type": 0,
                        }
                      }
                      strokeLinecap={1}
                      strokeLinejoin={1}
                      strokeWidth="2"
                    >
                      <RNSVGCircle
                        cx="12.1"
                        cy="12.1"
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                        propList={
                          [
                            "stroke",
                          ]
                        }
                        r="1"
                        stroke={
                          {
                            "payload": 4282448720,
                            "type": 0,
                          }
                        }
                      />
                    </RNSVGGroup>
                  </RNSVGSvgView>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
      <View
        style={
          {
            "backgroundColor": "#111f22",
            "borderBottomLeftRadius": 0,
            "borderBottomRightRadius": 0,
            "borderBottomWidth": 0,
            "borderLeftWidth": 0,
            "borderRightWidth": 0,
            "borderStyle": "solid",
            "borderTopLeftRadius": 0,
            "borderTopRightRadius": 0,
            "borderTopWidth": 0,
            "bottom": "-100%",
            "flex": 1,
            "flexDirection": "column",
            "height": 0,
            "left": 0,
            "maxHeight": "100%",
            "maxWidth": 738,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "shadowColor": undefined,
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
            "width": "100%",
            "zIndex": -1,
          }
        }
      />
    </View>
  </View>,
]
`;
